prow/plank/reconciler.go

```diff
@@ -344,12 +344,23 @@ func (r *reconciler) defaultReconcile(ctx context.Context, request reconcile.Req
 		// Objects can be deleted from the API while being in our workqueue
 		return reconcile.Result{}, nil
 	}
+	originalPJ := pj.DeepCopy()
 
 	res, err := r.serializeIfNeeded(ctx, pj)
 	if IsTerminalError(err) {
 		// Unfixable cases like missing build clusters, do not return an error to prevent requeuing
-		r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj)).
-			Error("Reconciliation failed with terminal error and will not be requeued")
+		log := r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj))
+		log.Error("Reconciliation failed with terminal error and will not be requeued")
```

Comment:
Is this message always true, or only if the Patch succeeds below?

I wonder if it it is better logged as the else message on the Patch() instead?

Reply:
This is always true because we return the same result (which does not requeue) whether or not the patch succeeds or fails.